packages/block-library/src/image/deprecated.js

```diff
@@ -8,353 +8,535 @@ import classnames from 'classnames';
-					height={ height }
-					title={ title }
-				/>
-			);
+		const classes = classnames( {
+			[ `align${ align }` ]: align,
+			[ `size-${ sizeSlug }` ]: sizeSlug,
+			'is-resized': width || height,
+		} );
 
-			const figure = (
-				<>
-					{ href ? (
-						<a
-							className={ linkClass }
-							href={ href }
-							target={ linkTarget }
-							rel={ newRel }
-						>
-							{ image }
-						</a>
-					) : (
-						image
-					) }
-					{ ! RichText.isEmpty( caption ) && (
-						<RichText.Content
-							tagName="figcaption"
-							value={ caption }
-						/>
-					) }
-				</>
-			);
+		const image = (
+			<img
+				src={ url }
+				alt={ alt }
+				className={ id ? `wp-image-${ id }` : null }
+				width={ width }
+				height={ height }
+				title={ title }
+			/>
+		);
 
-			if ( 'left' === align || 'right' === align || 'center' === align ) {
-				return (
-					<div { ...useBlockProps.save() }>
-						<figure className={ classes }>{ figure }</figure>
-					</div>
-				);
-			}
+		const figure = (
+			<>
+				{ href ? (
+					<a
+						className={ linkClass }
+						href={ href }
+						target={ linkTarget }
+						rel={ newRel }
+					>
+						{ image }
+					</a>
+				) : (
+					image
+				) }
+				{ ! RichText.isEmpty( caption ) && (
+					<RichText.Content tagName="figcaption" value={ caption } />
+				) }
+			</>
+		);
 
+		if ( 'left' === align || 'right' === align || 'center' === align ) {
 			return (
-				<figure { ...useBlockProps.save( { className: classes } ) }>
-					{ figure }
-				</figure>
+				<div { ...useBlockProps.save() }>
+					<figure className={ classes }>{ figure }</figure>
+				</div>
 			);
-		},
-	},
-	{
-		attributes: blockAttributes,
-		save( { attributes } ) {
-			const { url, alt, caption, align, href, width, height, id } =
-				attributes;
-
-			const classes = classnames( {
-				[ `align${ align }` ]: align,
-				'is-resized': width || height,
-			} );
+		}
 
-			const image = (
-				<img
-					src={ url }
-					alt={ alt }
-					className={ id ? `wp-image-${ id }` : null }
-					width={ width }
-					height={ height }
-				/>
-			);
+		return (
+			<figure { ...useBlockProps.save( { className: classes } ) }>
+				{ figure }
+			</figure>
+		);
+	},
+};
 
-			return (
-				<figure className={ classes }>
-					{ href ? <a href={ href }>{ image }</a> : image }
-					{ ! RichText.isEmpty( caption ) && (
-						<RichText.Content
-							tagName="figcaption"
-							value={ caption }
-						/>
-					) }
-				</figure>
-			);
+/**
+ * Deprecation for moving existing border radius styles onto the inner img
+ * element where new border block support styles must be applied.
+ * It will also add a new `.has-custom-border` class for existing blocks
+ * with border radii set. This class is required to improve caption position
+ * and styling when an image within a gallery has a custom border or
+ * rounded corners.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/31366
+ */
+const v5 = {
```

Comment:
See https://github.com/WordPress/gutenberg/blob/0bf26503cbea0f51f403107dcf6bcda856766b5f/packages/block-library/src/image/block.json for the definition at the time the v5 deprecation was added.